### ISSUE TITLE
searching for 'elvui' and clicking an addon causes window to freeze

### ISSUE THREAD

**Issue by torkus** [ogri-la/strongbox#402]
**Describe the bug**
*A clear and concise description of what the bug is.*

Opening any version of strongbox from 5.1.0 onwards (possibly even further back), clicking 'search', typing 'elvui' and then clicking an addon causes the window to freeze.

Doesn't affect my dev login, does affect my regular login, affects macs, doesn't seem to affect the current develop branch as of 2023-05-07 (6.2.0, unreleased).

**Comment by sergen213** [comment]
I was losing my mind. I though my Java installation was corrupted. I can reproduce the same error. Searching for Elvui and clicking an addon causes any windows to freeze, I have to wait until it gives not responding error. I'm on Nobara 37 both Appimage and Jar doing the same.

**Comment by torkus** [comment]
It's crazy it's gone undetected for so long! It just doesn't crop up in my dev environment and I'm at a loss for what it could be. I'll  have to add some automated GUI tests at some point.

**Comment by torkus** [comment]
This here is the trigger for the freezing:
https://github.com/ogri-la/strongbox/blob/b71dea5fdba01c10cbfc1f4dd2252154b4b0fe7a/src/strongbox/jfx.clj#L2020

But I still can't figure out what the actual problem is. I do know it was introduced in 5.1.0 so I'm able to diff the changes but it's taken some time just to get to this point so I'll release a 6.1.2 tomorrow disabling this while I work on finding a better solution.

**Comment by sergen213** [comment]
> This here is the trigger for the freezing:
> https://github.com/ogri-la/strongbox/blob/b71dea5fdba01c10cbfc1f4dd2252154b4b0fe7a/src/strongbox/jfx.clj#L2020
> 
> But I still can't figure out what the actual problem is. I do know it was introduced in 5.1.0 so I'm able to diff the changes but it's taken some time just to get to this point so I'll release a 6.1.2 tomorrow disabling this while I work on finding a better solution.

Maybe something related to tukui? Because it only happens with elvui. I'll try something else from tukui.

**Comment by torkus** [comment]
No, nothing to do with the catalogue and everything to do with the GUI I'm afraid.

It's also non-deterministic, which means some of the time it works! I've been going through all of the changes that were made and testing them three times each looking for a consistent three passes. Such a pita.

**Comment by sergen213** [comment]
I narrowed down the issue to Catalogues if you set your catalogue to full and search for it, it freezes but if I switch it to Short or others then search it's all fine. I can even switch to Full after searching with short and it's not freezing but if I do my first search with Full then it freezes. It might be line 2046 and below. I'm not a programmer myself but I think when it's in Full mode it tries to get known hosts and dbstats and something triggers this and it freezes or maybe I'm just wrong.

https://github.com/ogri-la/strongbox/blob/b71dea5fdba01c10cbfc1f4dd2252154b4b0fe7a/src/strongbox/jfx.clj#L2046

**Comment by torkus** [comment]
> I narrowed down the issue to Catalogues if you set your catalogue to full and search for it, it freezes but if I switch it to Short or others then search it's all fine. I can even switch to Full after searching with short and it's not freezing but if I do my first search with Full then it freezes.

This is interesting, I'll investigate it. Thanks for the sleuthing, it's appreciated.

**Comment by torkus** [comment]
`6.1.2` has been released with a fix in it.

I'm closing this issue now but please let me know if it doesn't work for you.
